indexer/road_shields_parser.cpp

```diff
@@ -0,0 +1,308 @@
+#include "indexer/road_shields_parser.hpp"
+
+#include "indexer/feature.hpp"
+
+#include "base/string_utils.hpp"
+
+#include <unordered_map>
+#include <utility>
+
+namespace ftypes
+{
+class RoadShieldParser
+{
+public:
+  RoadShieldParser(std::string const & baseRoadNumber)
+  : m_baseRoadNumber(baseRoadNumber)
+  {}
+  virtual ~RoadShieldParser(){}
+  virtual RoadShield ParseRoadShield(std::string const & rawText) = 0;
+
+  std::vector<RoadShield> GetRoadShields()
+  {
+    std::vector<RoadShield> result;
+    std::vector<std::string> shieldsRawTests = strings::Tokenize(m_baseRoadNumber, ";");
+    for (std::string const & rawText : shieldsRawTests)
+    {
+      RoadShield shield = ParseRoadShield(rawText);
+      if (!shield.m_name.empty())
+        result.push_back(std::move(shield));
+    }
+    return result;
+  }
+
+protected:
+  std::string const m_baseRoadNumber;
+};
+
+uint32_t constexpr kMaxRoadShieldTextSize = 8;
+
+class USRoadShieldParser : public RoadShieldParser
+{
+public:
+  USRoadShieldParser(std::string const & baseRoadNumber)
+  : RoadShieldParser(baseRoadNumber)
+  {}
+
+  RoadShield ParseRoadShield(std::string const & rawText) override
+  {
+    static std::vector<std::string> kStatesCode = {
```

Comment:
Looks like it will be better if I replace it with std::array and move it into a class scope to avoid guards which are produced by local static constants.